crates/autopilot/src/decoded_settlement.rs

```diff
@@ -331,8 +331,8 @@ impl DecodedSettlement {
         let sell_token = self.tokens.get(sell_index)?;
         let buy_token = self.tokens.get(buy_index)?;
 
-        let (kind, fee) = match trade.fee_amount.is_zero() {
-            false => (FeeKind::User, trade.fee_amount),
+        let (kind, fee, surplus_token) = match trade.fee_amount.is_zero() {
```

Comment:
Might be slightly nicer to read the code if the surplus token gets decided in a separate match statement:
```rust
        let surplus_token = match (trade.fee_amount.is_zero(), trade.flags.order_kind()) {
            (false, _) => sell_token, // pre-signed fee is always in the sell token
            (_, OrderKind::Buy) => sell_token,
            (_, OrderKind::Sell) => buy_token,
        };
```